Orange/data/variable.py

```diff
@@ -54,7 +54,9 @@ def is_discrete_values(values):
             return False
 
     # Strip NaN from unique
-    unique = {i for i in unique if not (isinstance(i, Number) and np.isnan(i))}
+    unique = {i for i in unique
+              if (not i in MISSING_VALUES and
```

Comment:
This was not necessary, but only because this check is made in io already. In other situations, it will prove more robust to repeat it here.